### ISSUE TITLE
Extend check functionality of conf.py option needs_sphinx

### ISSUE THREAD

**Issue by fkromer** [sphinx-doc/sphinx#2235]
The option `needs_sphinx` in _conf.py_ does support the major and the minor version number e.g. `needs_sphinx = '1.3'` for Sphinx v1.3.3, Sphinx v1.3.4, and so on (tested with Sphinx v1.3.4). In some situations (needs v1.3.4 and shall warn when v1.3.3 is used) it would be helpful to have the whole Sphinx version checked.

**Comment by tk0miya** [comment]
It was rejected at #527. But I think it is no reason to ignore micro version.
There is no demerit if we support it.

So I will implement this if no objection.

**Comment by fkromer** [comment]
I posted the issue because i came across an integration issue of my generated .html files (without issue on localhost) which are hosted on readthedocs.org. My sphinx-doc environment requires Sphinx v1.3.4 which i can guarantee on my local environment. However it seems like readthedocs.org uses another Sphinx v1.3.x version as default (conf.py: needs_sphinx = 1.3). The .html files where build but without supporting a feature of my (local) website. If the option needs_sphinx in conf.py would specify the whole version the website build of readthedocs.org should have failed because conf.py is used by readthedocs.org for building.

Hint: I was able to force readthedocs.org to use Sphinx v1.3.4 by specifying to build from virtualenv and by specifying the corresponding requirements.txt file in the readthedocs.org project settings (project/advanced settings/"Install your project inside a virtualenv using setup.py install" & path from project root to pip requirements file).
